modules/social_features/social_out_of_office/social_out_of_office.module

```diff
@@ -0,0 +1,270 @@
+    if ($parent_comment) {
+      $parent_comment_author = $parent_comment->getOwner();
+      /** @var \Drupal\profile\Entity\Profile $profile */
+      $profile = $profile_storage->loadByUser($parent_comment_author, SocialOutOfOfficeHelperInterface::PROFILE_TYPE);
+      $post_comment_related_profiles[] = $profile;
+    }
+  }
+  // Post entity type.
+  else {
+    $body_value = $entity->get('field_post')->getValue();
+  }
+
+  /** @var \Drupal\social_out_of_office\SocialOutOfOfficeHelperInterface $ooo_helper */
+  $ooo_helper = \Drupal::service('social_out_of_office.helper');
+
+  /** @var \Drupal\social_mentions\SocialMentionsHelperInterface $social_mentions_helper */
+  $social_mentions_helper = \Drupal::service('social_mentions.helper');
+
+  // Prepare body value.
+  $body_value = $body_value[0]['value'] ?? '';
+
+  // Retrieve mentions from body.
+  $mentions = $social_mentions_helper->getMentions($body_value);
+
+  // Go through all mentions.
+  if (!empty($mentions)) {
+    foreach ($mentions as $id) {
+      /** @var \Drupal\profile\Entity\Profile $profile */
+      $profile = $profile_storage->load($id);
+
+      if (!($profile instanceof ProfileInterface)) {
+        continue;
+      }
+
+      // Remove mention profile if it is in related profiles array.
+      if (($index = array_search($profile, $post_comment_related_profiles)) !== FALSE) {
+        unset($post_comment_related_profiles[$index]);
+      }
+
+      // Show out of office message that contains in user profile.
+      $ooo_helper->showOutOfOfficeStatusMessage($profile);
+    }
+  }
+
+  // Show OoO messages for related profiles.
+  foreach ($post_comment_related_profiles as $post_comment_related_profile) {
+    // Show out of office message that contains in user profile.
+    $ooo_helper->showOutOfOfficeStatusMessage($post_comment_related_profile);
+  }
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function social_out_of_office_form_private_message_add_form_alter(array &$form, FormStateInterface $form_state, string $form_id): void {
+  // Make sure that we add custom submit handler exactly for create new chat
+  // with user since we already add message about out of office for existing
+  // chats.
+  // @see social_out_of_office_preprocess_user()
+  if (\Drupal::routeMatch()->getRouteName() === 'private_message.private_message_create') {
+    $form['actions']['submit']['#submit'][] = 'social_out_of_office_add_notice_about_users_ooo';
+  }
+}
+
+/**
+ * Submit function for 'form_private_message_add' form.
+ */
+function social_out_of_office_add_notice_about_users_ooo(array &$form, FormStateInterface $form_state): void {
+  /** @var \Drupal\social_out_of_office\SocialOutOfOfficeHelperInterface $ooo_helper */
+  $ooo_helper = \Drupal::service('social_out_of_office.helper');
+
+  // Check if any of members is out of office and add status message if yes.
+  $members = $form_state->getValue('members');
+  foreach ($members as $member) {
```

Comment:
this is not really performant, imagine a thread with over 100 people. Best would be to write an alternative here, maybe using a loadMultiple or potentially even a custom query that accepts multiple ids and returns their message

Reply:
Agree. I've improved this.